### DISCUSSION TITLE
Run Plugin from Template? AI Specifically

### DISCUSSION THREAD

**Discussion by chrismalek** [NotePlan/plugins#433] [Q&A]
Is there a way to run a plugin command from a template?

I want in my daily note to have ChatGPT create a random journaling prompt. I have been playing around with the AI plugin but I don't see a way in the documentation to trigger that plugin from a template.  

So after I trigger the daily note template, I want the ChatGPT output to just give me a random journaling prompt in my template.

**Comment by dwertheimer** [comment]
Yes you can run a plug-in from a template. I will give some thought yo te best way to do what you are asking for here. To give me an example, what is the prompt you are using to generate the daily prompt? It might be much faster and much more efficient to ask chat GPT once to give you 100 prompts and then randomly choose from them each day. That would allow you to pull out the ones that are dumb. But I like the idea and direction as an intellectual exercise and so I will take it on later today.

**Comment by chrismalek** [comment]
I actually want the complete randomness of what OpenAI may return. I also may want to use it for some other use cases like generating fake inspirational quotes or bad dad jokes daily for my own amusement. I was staring with a journaling prompt.  

Simple Chat GPT Input for now --> `Generate a journaling prompt`

Here is a sample I just got back from GPT4.

> Reflect on a time when you faced a challenge or obstacle in your life, and describe the actions you took to overcome it. What were your emotions and thoughts throughout the process? How did this experience contribute to your personal growth, and what lessons did you learn from it that you can apply to future challenges?

**Reply by dwertheimer** [reply]
@chrismalek Been working on this. Found a long-standing bug in Templating that caused Templating to get confused if the word "prompt" occurred in a tag. That was not obvious and had me running around in circles for awhile. Now can get back to working on the AI piece of it. Interestingly, ChatGPT continues to send me the same prompt frequently, so when I release it, you may need to play around with your prompt a bit to try to introduce some randomness. In the meantime, pls update your Templating plugin.

**Comment by chrismalek** [comment]
Yes I saw you push that change I am watcher on the GitHub project and I had a feeling that push was related to this. 

Do you have a rough template example of calling the chatgpt / ai plugin from the template that you can paste in here? I am struggling finding an example of calling plugins from a template.  I am javascript challenged :-). I am sure I am just looking in the wrong place.

**Reply by dwertheimer** [reply]
I'm still working on this. Needed to fix Templating first

**Comment by dwertheimer** [comment]
@chrismalek Just released v0.7.0-beta1 of the NotePlan AI plugin. Pls have a try. 

Here's the [documentation](https://github.com/NotePlan/plugins/blob/main/shared.AI/README.md#one-shot-calling-from-x-callback-link-or-via-templating-tag) for the templating command:

**Comment by chrismalek** [comment]
@dwertheimer Awesome..it works Thank you so much.  Long live Noteplan (and all the plugins). 

👏🏻  😄

**Reply by dwertheimer** [reply]
😀
